apps/web/src/pages/integrations/components/multi-provider/SelectProviderSidebar.tsx

```diff
@@ -126,36 +129,20 @@ export function SelectProviderSidebar() {
         </Tabs>
         <Space h={20} />
         <CenterDiv>
-          <ListProviders
-            selectedProvider={selectedProvider}
-            onProviderClick={onProviderClick}
-            channelProviders={inAppProviderList}
-            channelType={ChannelTypeEnum.IN_APP}
-          />
-          <ListProviders
-            selectedProvider={selectedProvider}
-            onProviderClick={onProviderClick}
-            channelProviders={emailProviderList}
-            channelType={ChannelTypeEnum.EMAIL}
-          />
-          <ListProviders
-            selectedProvider={selectedProvider}
-            onProviderClick={onProviderClick}
-            channelProviders={chatProviderList}
-            channelType={ChannelTypeEnum.CHAT}
-          />
-          <ListProviders
-            selectedProvider={selectedProvider}
-            onProviderClick={onProviderClick}
-            channelProviders={pushProviderList}
-            channelType={ChannelTypeEnum.PUSH}
-          />
-          <ListProviders
-            selectedProvider={selectedProvider}
-            onProviderClick={onProviderClick}
-            channelProviders={smsProviderList}
-            channelType={ChannelTypeEnum.SMS}
-          />
+          <When truthy={Object.values(providersList).every((list) => list.length === 0)}>
+            <Image src={`${CONTEXT_PATH}/static/images/empty-provider-search.png`} />
+          </When>
+          {CHANNELS_ORDER.map((channelType) => {
+            return (
+              <ListProviders
+                key={channelType}
+                selectedProvider={selectedProvider}
+                onProviderClick={onProviderClick}
+                channelProviders={providersList[channelType]}
+                channelType={channelType}
+              />
+            );
```

Comment:
when no providers, we don't have to render these at all... rn they are hidden, but we do waste this way react time and memory resources